litespi/clkgen.py

```diff
@@ -104,6 +104,11 @@ def __init__(self, pads, device, cnt_width=8, with_ddr=False):
                 # startupe2 needs 3 usrcclko cycles to switch over to user clock
                 self.comb += en_int.eq(cycles < 3)
                 self.sync += If(en_int & posedge, cycles.eq(cycles+1))
+            elif device[:5] == "LFE5U":
```

Comment:
Wouldn't it be better to use `device.startswith("LFE5U")` instead of directly providing range and comparing?
We also should modify the condition for using `STARTUPE2` to also use `device.startswith("xc7")`

Reply:
That is better! Also modified STARTUPE2 condition as suggested.